### PR TITLE
Flash fix

### PR DESCRIPTION
**Pull request by JakubBrachTieto**
FIX for Flash problems.
Platform crash on Flash Write - FIXED
Platform crash on Flash Erase - FIXED
factoryreset functionlaity        - FIXED

### CODE REVIEW

**Comment by jwhui** [comment]
After #1250 is merged, please rebase this PR to remove 91adb61 from the commit history.

**Comment by jwhui** [comment]
Please rebase to resolve merge conflicts.

File: examples/platforms/da15000/radio.c [hunk 1]
```diff
@@ -31,528 +31,344 @@
...
-    volatile uint32_t *txFlagClearM;
-    txFlagClearM   = FTDF_GET_FIELD_ADDR_INDEXED(ON_OFF_REGMAP_TX_FLAG_CLEAR_M, FTDF_TX_DATA_BUFFER);
-    *txFlagClearM |= MSK_F_FTDF_ON_OFF_REGMAP_TX_FLAG_CLEAR_M;
-    txFlagClearM   = FTDF_GET_FIELD_ADDR_INDEXED(ON_OFF_REGMAP_TX_FLAG_CLEAR_M, FTDF_TX_WAKEUP_BUFFER);
-    *txFlagClearM |= MSK_F_FTDF_ON_OFF_REGMAP_TX_FLAG_CLEAR_M;
-}
-
+    (void)aInstance;
 
+    aIeeeEui64[0] = 0x18;
```
Comment (aeliot) [inline examples/platforms/da15000/radio.c]: This dialog code should not be using the Nest Labs OUI.

File: examples/platforms/da15000/radio.c [hunk 2]
```diff
@@ -31,528 +31,344 @@
...
+
     return kThreadError_None;
 }
 
 bool otPlatRadioIsEnabled(otInstance *aInstance)
 {
     (void)aInstance;
-    return (s_state != kStateDisabled);
+
+    return (RadioState != kStateDisabled) ? true : false;
```
Comment (aeliot) [inline examples/platforms/da15000/radio.c]: You can simply return ``(RadioState != kStateDisabled)`` without the inline if statement.

File: examples/platforms/da15000/radio.c [hunk 3]
```diff
@@ -31,528 +31,344 @@
...
-    ThreadError error = kThreadError_None;
     (void)aInstance;
 
-#if DEBUG_LOG_ENABLE
-    hw_uart_write_buffer(HW_UART1, "\nR", 2);
-#endif
-    VerifyOrExit(s_state != kStateDisabled, error = kThreadError_Busy);
-    s_state = kStateReceive;
-    s_receive_frame.mChannel = aChannel;
+    uint8_t EnableRX;
```
Comment (aeliot) [inline examples/platforms/da15000/radio.c]: Local variables shouldn't be capitalized.

File: examples/platforms/da15000/flash.c [hunk 4]
```diff
@@ -46,27 +44,24 @@
...
 {
 
-    uint32_t flash_offset = (aAddress + FLASH_OFFSET) & ~(FLASH_SECTOR_SIZE - 1);
+    uint32_t flash_offset = (aAddress) & ~(FLASH_SECTOR_SIZE - 1);
 
-    if (aAddress > FLASH_BUFFER_SIZE)
-    {
-        return kThreadError_InvalidArgs;
-    }
+    if (DisableErase) { return kThreadError_None; }
```
Comment (aeliot) [inline examples/platforms/da15000/flash.c]: Could you separate this ``if`` statement out into multiple lines.

**Pull request merged**